Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -2220,30 +2225,41 @@ void RenderFlexibleBox::alignChildren(const Vector<LineContext>& lineContexts)
                 continue;
             
             ItemPosition position = alignmentForChild(flexItem.box);
+            LayoutUnit maxAscent = position == ItemPosition::LastBaseline ? lineContext.lastBaselineMaxAscent : lineContext.maxAscent;
             if (position == ItemPosition::Stretch)
                 applyStretchAlignmentToChild(flexItem.box, lineCrossAxisExtent);
             LayoutUnit availableSpace = availableAlignmentSpaceForChild(lineCrossAxisExtent, flexItem.box);
-            LayoutUnit offset = alignmentOffset(availableSpace, position, marginBoxAscentForChild(flexItem.box), maxAscent, style().flexWrap() == FlexWrap::Reverse);
+            LayoutUnit offset = alignmentOffset(availableSpace, position, marginBoxAscentForChild(flexItem.box), maxAscent, style().flexWrap() == FlexWrap::Reverse, childIsParticipatingInBaselineAlignment(flexItem.box));
             adjustAlignmentForChild(flexItem.box, offset);
-            if (position == ItemPosition::Baseline && style().flexWrap() == FlexWrap::Reverse)
+            if ((position == ItemPosition::Baseline && containerHasWrapReverse) || (childIsParticipatingInBaselineAlignment(flexItem.box) && position == ItemPosition::LastBaseline && !containerHasWrapReverse)) {
+                needToAdjustItemsTowardsCrossAxis = true;
                 minMarginAfterBaseline = std::min(minMarginAfterBaseline, availableAlignmentSpaceForChild(lineCrossAxisExtent, flexItem.box) - offset);
+            }
         }
 
         minMarginAfterBaselines.append(minMarginAfterBaseline);
     }
-    
-    if (style().flexWrap() != FlexWrap::Reverse)
+
+    if (!needToAdjustItemsTowardsCrossAxis)
         return;
-    
+
+    auto adjustWrapReverseItem = [this, containerHasWrapReverse](RenderBox& box) { 
```

Comment:
This doesn't seem to be adjusting anything, so rename to `needToAdjustWrapReverseItem`, `shouldAdjustWrapReverseItem`, or similar.
And `box` can be `const`.